public/app/components/codeEditor/codeEditor.js

```diff
@@ -27,13 +27,29 @@ angular.module('kibibitCodeEditor')
     SettingsService) {
 
     var vm = this;
+    var editorSettings = SettingsService.settings.editorSettings;
+    var defaultEditorSettings = {
+      'wrap': editorSettings.lineWrap,
+      'mode': 'ace/mode/' + editorSettings.syntaxMode,
+      'theme': 'ace/theme/' + editorSettings.theme,
+      'tabSize': editorSettings.tabWidth,
+      'fontSize': editorSettings.fontSize,
+      'showGutter': editorSettings.isGutter,
+      'useSoftTabs': editorSettings.isSoftTabs,
+      'showPrintMargin': editorSettings.ruler
+    };
+
+    var initCodeEditor = function(editor, settings) {
```

Comment:
not exactly what I meant :-)

I thought of something like this:

```javascript
var init = function(settings) {
  editor.setOptions({
    'wrap': settings.lineWrap,
    'mode': 'ace/mode/' + settings.syntaxMode,
    'theme': 'ace/theme/' + settings.theme,
    'tabSize': settings.tabWidth,
    'fontSize': settings.fontSize,
    'showGutter': settings.isGutter,
    'useSoftTabs': settings.isSoftTabs,
    'showPrintMargin': settings.ruler
  });
}
```

Expose the editor as a private varialbe (each instance of the `kb-code-editor` should know about it's own `editor` object.

Just add a variable and update it on `vm.aceLoaded`